WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/history/ScanHistoryListViewModelTest.kt

```diff
@@ -34,16 +43,54 @@ class ScanHistoryListViewModelTest {
     @Before
     fun setUp() = test {
         viewModel = ScanHistoryListViewModel(scanThreatItemBuilder, TEST_DISPATCHER)
-        whenever(scanThreatItemBuilder.buildThreatItem(anyOrNull(), anyOrNull())).thenReturn(mock())
+        whenever(scanHistoryViewModel.threats).thenReturn(
+                MutableLiveData(
+                        listOf(
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.FIXED)),
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.UNKNOWN)),
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.FIXED)),
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.IGNORED)),
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.FIXED)),
+                                GenericThreatModel(genericThreatModel.baseThreatModel.copy(status = ThreatStatus.CURRENT))
+                        )
+                )
+        )
     }
 
     @Test
     fun `Threat ui state items shown, when the data is available`() {
-        whenever(scanHistoryViewModel.threats).thenReturn(MutableLiveData(listOf(mock(), mock())))
-
-        viewModel.start(site, scanHistoryViewModel)
+        viewModel.start(ScanHistoryTabType.ALL, site, scanHistoryViewModel)
         viewModel.uiState.observeForever(mock())
 
         assertThat(viewModel.uiState.value).isNotEmpty
     }
+
+    @Test
+    fun `Only fixed threats are shown, when fixed tab is selected`() {
+        viewModel.start(ScanHistoryTabType.FIXED, site, scanHistoryViewModel)
+        viewModel.uiState.observeForever(mock())
+
+        verify(scanThreatItemBuilder, atLeast(1)).buildThreatItem(captor.capture(), anyOrNull())
```

Comment:
Idea (💡): What if instead of `atLeast(1)` you used a more explicit verification, like `times(3)`? This way the test becomes more solid as it will fail if you add/remove `FIXED` threats from the list or change the filtering logic.

PS: The same applies for the other two similar test below.

Reply:
Actually I had `times(3)` there and changed it to `atLeast(1)` to make it more solid:D. If you add/remove threats into the list, the test will still work without changes and it'll still verify that only "FIXED" threats are returned (unless you remove all FIXED threats in which case it'll fail, but that's good because it won't be able to verify the behavior in that case). However, we'll need to update the test every time we touch the threat list if we use "times(3)".

Can you please elaborate how it becomes more solid with `times(3)`?

Reply:
Ohh I see, you want the test to verify that all FIXED threats passed the filtering and non of them was filter out even when it shouldn't. I'll update it with "times(x)", I guess it makes sense.

Reply:
Updated in https://github.com/wordpress-mobile/WordPress-Android/pull/13827/commits/06744f918991f3ee1058587ae0cd522316938eb0

Reply:
Yes exactly your last comment Jirka! 💪 

Also, with `times(x)` if the main code filtering logic changes, say filtering by `FIXED` and something else, with the test list being untouched, then the test might still pass and produce a false positive, then we would need to update our tests since the main code got updated as well.

The solid/robust approach you were thinking is correct, but in this case, because it is applied to having the test more robust because of the test being updated (aka. adding/removing threats on the test list), then this doesn't provide much value. If the test constant is changing (in this case the test list), it is expected for the tests to change with it as well. Making that more solid/robust might create a problem with the production code.

I know, it is a thin line and I keep fighting with that line every single time! 😄

Reply:
I totally agree, I tried to decrease the maintenance cost of the unit tests but I didn't realize it decreases the value of the test. Definitely a good call from you, thanks for the suggestion and elaborate reply!